azurerm/internal/services/network/network_interface_network_security_group_association_resource.go

```diff
@@ -0,0 +1,218 @@
+)
+
+func resourceArmNetworkInterfaceSecurityGroupAssociation() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmNetworkInterfaceSecurityGroupAssociationCreate,
+		Read:   resourceArmNetworkInterfaceSecurityGroupAssociationRead,
+		Delete: resourceArmNetworkInterfaceSecurityGroupAssociationDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"network_interface_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: azure.ValidateResourceID,
+			},
+
+			"network_security_group_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: azure.ValidateResourceID,
+			},
+		},
+	}
+}
+
+func resourceArmNetworkInterfaceSecurityGroupAssociationCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Network.InterfacesClient
+	ctx, cancel := timeouts.ForCreate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	log.Printf("[INFO] preparing arguments for Network Interface <-> Network Security Group Association creation.")
+
+	networkInterfaceId := d.Get("network_interface_id").(string)
+	networkSecurityGroupId := d.Get("network_security_group_id").(string)
+
+	nicId, err := azure.ParseAzureResourceID(networkInterfaceId)
+	if err != nil {
+		return err
+	}
+
+	networkInterfaceName := nicId.Path["networkInterfaces"]
+	resourceGroup := nicId.ResourceGroup
+
+	locks.ByName(networkInterfaceName, networkInterfaceResourceName)
+	defer locks.UnlockByName(networkInterfaceName, networkInterfaceResourceName)
+
+	nsgId, err := azure.ParseAzureResourceID(networkSecurityGroupId)
+	if err != nil {
+		return err
+	}
+	nsgName := nsgId.Path["networkSecurityGroups"]
+
+	locks.ByName(nsgName, networkSecurityGroupResourceName)
+	defer locks.UnlockByName(nsgName, networkSecurityGroupResourceName)
+
+	read, err := client.Get(ctx, resourceGroup, networkInterfaceName, "")
+	if err != nil {
+		if utils.ResponseWasNotFound(read.Response) {
+			return fmt.Errorf("Network Interface %q (Resource Group %q) was not found!", networkInterfaceName, resourceGroup)
+		}
+
+		return fmt.Errorf("Error retrieving Network Interface %q (Resource Group %q): %+v", networkInterfaceName, resourceGroup, err)
+	}
+
+	props := read.InterfacePropertiesFormat
+	if props == nil {
+		return fmt.Errorf("Error: `properties` was nil for Network Interface %q (Resource Group %q)", networkInterfaceName, resourceGroup)
+	}
+
+	// first double-check it doesn't exist
+	resourceId := fmt.Sprintf("%s|%s", networkInterfaceId, networkSecurityGroupId)
```

Comment:
The pipe separator may cause issue with `terraform import` ? (see comment on docs)